core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -425,8 +434,25 @@ private void tryStart(Instant startedAt) {
             // For all registered output consumers, start following as close to container startup as possible
             this.logConsumers.forEach(this::followOutput);
 
+            // Wait until inspect container returns the mapped ports
+            Callable<Integer> inspectMappedExposedPortsCount = () -> {
+                containerInfo = dockerClient.inspectContainerCmd(containerId).exec();
+                return (int) containerInfo
+                    .getNetworkSettings()
+                    .getPorts()
+                    .getBindings()
+                    .values()
+                    .stream()
+                        .filter(Objects::nonNull)
+                        .count();
+            };
+
+            await()
+                .atMost(5, SECONDS)
+                .pollInterval(fibonacci(MILLISECONDS))
```

Comment:
I'd be inclined to stick with a fixed poll interval here. Fibonacci backoff is important when protecting a shared resource from a thundering herd, but here there's only one caller so we don't gain anything by backing off in big steps.

Reply:
But this means we could have just kept the `Unreliables` implementation using the `Thread.sleep()`, or?

Reply:
It looks like we're missing https://github.com/awaitility/awaitility/issues/214 and for some reason I thought Awaitility has it already 😅